irods/test/meta_test.py

```diff
@@ -520,13 +520,41 @@ def units():
                 if d: d.unlink(force = True)
                 helpers.remove_unused_metadata(session)
 
+    def test_AVUs_populated_improperly_with_empties_or_nonstrings_fail_identically__issue_547(self):
+        try:
+            to_delete = []
+            hc = helpers.home_collection(self.sess)
+            mtemplate = iRODSMeta('some_name', 'some_value', 'some_units')
+            for index in ('name','value','units'):
+                for edge_case_arg in ('', 3):
+                    # Empty units are permitted, ie. iRODSMeta(attr,value,'') and iRODSMeta(attr,value) are equivalent.
+                    if edge_case_arg in ('',b'') and index == 'units':
+                        continue
+                    for method in ('set','add'):
+                        data = self.sess.data_objects.create("{hc}/{index}_{edge_case_arg}_{method}_AZ__issue_547".format(**locals()))
```

Comment:
Do we need to create a new data object for each iteration of the loop?

Reply:
We don't have to, no.  But it is cheap, and it avoids having to clear the metadata previously set on a  object if maintaining the same one object over all iterations.